libyul/backends/evm/StackLayoutGenerator.cpp

```diff
@@ -549,6 +620,55 @@ Stack StackLayoutGenerator::combineStack(Stack const& _stack1, Stack const& _sta
 	return commonPrefix + bestCandidate;
 }
 
+vector<StackLayoutGenerator::StackTooDeep> StackLayoutGenerator::reportStackTooDeep(CFG::BasicBlock const& _entry) const
+{
+	vector<StackTooDeep> stackTooDeepErrors;
+	util::BreadthFirstSearch<CFG::BasicBlock const*> breadthFirstSearch{{&_entry}};
+	breadthFirstSearch.run([&](CFG::BasicBlock const* _block, auto _addChild) {
+		Stack stack;
+		stack = m_layout.blockInfos.at(_block).entryLayout;
+
+		for (auto const& operation: _block->operations)
+		{
+			Stack& operationEntry = m_layout.operationEntryLayout.at(&operation);
+
+			stackTooDeepErrors += findStackTooDeep(stack, operationEntry);
+			stack = operationEntry;
+			for (size_t i = 0; i < operation.input.size(); i++)
+				stack.pop_back();
+			stack += operation.output;
+		}
+		// Do not attempt to create the exit layout here, since the code generator will directly move to the
+		// target entry layout.
```

Comment:
I do not understand this comment. Don't we have the exit layout already created in `stack` variable at this point?

Reply:
I made the comment more precise - it was meant to make it clear that not considering ``m_layout.blockInfos.at(_block).exitLayout`` is on purpose.

Reply:
Ah, the function is supposed to generate `m_layout.blockInfos.at(_block).exitLayout`?

Well, the confusing part for me was that gathering info about stack too deep errors happens on the fly, while the stack layout is being generated, but the added code is not all that much different from one that would be doing it after the fact. It only shows through in a few places and at some point I was even wondering why some of these functions are not `const` :)

I guess this is caused by the fact that I'm reviewing the diff rather than at the class as a whole so I'm only aware of how that reviewed part works. Still, despite problems like this, I think these smaller, more focused PRs make reviewing easier overall.